File: Readio/src/main/java/ninegle/Readio/book/dto/BookResponseDto.java
package ninegle.Readio.book.dto;

import java.time.LocalDate;

import lombok.Builder;
import ninegle.Readio.book.dto.author.AuthorDto;
import ninegle.Readio.category.dto.CategoryDto;
import ninegle.Readio.publisher.dto.PublisherDto;

@Builder
public record BookResponseDto(long id, String name, String description, String image, String isbn, String ecn,
							  LocalDate pubDate, CategoryDto category, PublisherDto publisher, AuthorDto author) {

}

File: Readio/src/main/java/ninegle/Readio/book/dto/BookResponseDto.kt
package ninegle.Readio.book.dto

import lombok.Builder
import ninegle.Readio.book.dto.author.AuthorDto
import ninegle.Readio.category.dto.CategoryDto
import ninegle.Readio.publisher.dto.PublisherDto
import java.time.LocalDate

@Builder
data class BookResponseDto(
    val id: Long,
    val name: String,
    val description: String,
    val image: String,
    val isbn: String?,
    val ecn: String?,
    val pubDate: LocalDate,
    val category: CategoryDto,
    val publisher: PublisherDto,
    val author: AuthorDto
)
